peakina/readers/csv.py

```diff
@@ -0,0 +1,79 @@
+"""
+Module to add csv support
+"""
+from typing import List, Optional
+
+import pandas as pd
+
+# The chunksize value for previews
+PREVIEW_CHUNK_SIZE = 1024
+
+
+def _extract_columns(filepath: str, encoding: str, sep: str) -> List[str]:
+    with open(filepath, buffering=10000, encoding=encoding) as ff:
+        return ff.readline().replace("\n", "").split(sep)
+
+
+def read_csv(
+    filepath: str,
+    *,
+    sep: str = ",",
+    keep_default_na: bool = False,
+    encoding: str = "utf-8",
+    preview_offset: Optional[int] = None,
+    preview_nrows: Optional[int] = None,
+    chunksize: Optional[int] = None,
+    nrows: int = 500,
+    error_bad_lines: bool = False,
+    skiprows: Optional[int] = None,
+) -> pd.DataFrame:
+    """
+    The read_csv method is able to make a preview by reading on chunks
+
+    """
+    if preview_nrows is not None and preview_offset is not None:
+        if preview_offset == 0:
+            preview_offset = 1  # skip header
+        chunks = pd.read_csv(
+            filepath,
+            sep=sep,
+            header=None,
+            names=_extract_columns(filepath, encoding, sep),
+            keep_default_na=keep_default_na,
+            encoding=encoding,
+            nrows=preview_nrows,
+            skiprows=lambda idx: idx < preview_offset + (skiprows or 0),
```

Comment:
> **skiprows:list-like, int or callable, optional**
>   Line numbers to skip (0-indexed) or number of lines to skip (int) at the start of the file.
>    If callable, the callable function will be evaluated against the row indices, returning True if the row should be skipped and False otherwise. An example of a valid callable argument would be lambda x: x in [0, 2].

Why using a lambda and not "just" the number of row to skip from the start ? Does this have something to do with other options ? Does this have any impact on performance when preview pages at the end ?